e2e/tests/uniswap_trade_test.rs

```diff
@@ -130,7 +130,7 @@ async fn test_with_ganache() {
             .await
             .expect("Couldn't query domain separator"),
     );
-    let db = Database::new("postgresql://").unwrap();
+    let db = Arc::new(Database::new("postgresql://").unwrap());
```

Comment:
We can clone and pass around `Db` directly without wrapping it in `Arc` because it is using that internally already.

Reply:
I will give this a try again, but the first time I tried this there was something the compiler didn't like.

Reply:
This is going to cause a chain of reverts... I had to update the Event Updater to take Arc<Database> and this is why this unit test changed. Are you sure about this?

Reply:
Or do you mean that I should just change the unit test to use something like `Arc::from(db.clone())` everywhere?

Reply:
I mean that everything that you made store Arc<Db> could be just Db and we clone the Db directly. Don't see why it would cause an error because Db is Clone.

Reply:
Its all good now.